### PR TITLE
test(keycloak): add tests

### PR DESCRIPTION
**Pull request by schultzp2020**
image

### CODE REVIEW

**Comment by tanoggy** [comment]
skip test

**Comment by schultzp2020** [comment]
The remaining sonarcloud bugs and hotspots are false positives.

File: plugins/keycloak-backend/src/lib/read.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import KcAdminClient from '@keycloak/keycloak-admin-client';
+
+import { KeycloakAdminClientMock } from '../../__fixtures__/helpers';
+import { KeycloakProviderConfig } from './config';
+import { readKeycloakRealm } from './read';
+
+const config = {
+  realm: 'myrealm',
+} as unknown as KeycloakProviderConfig;
```
Comment (invincibleJai) [inline plugins/keycloak-backend/src/lib/read.test.ts:9]: nit: do we need to case as unknown? Can directly do `as KeycloakProviderConfig` or better to add required type with mock

```
const config: KeycloakProviderConfig = {
  realm: 'myrealm',
  id: 'mock_id',
  baseUrl: 'http://mock-url',
};
```

File: plugins/keycloak-backend/__fixtures__/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { EntityProviderConnection } from '@backstage/plugin-catalog-backend';
```
Comment (invincibleJai) [inline plugins/keycloak-backend/__fixtures__/helpers.ts:1]: @deprecated — import from @backstage/plugin-catalog-node instead

File: plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.test.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    const keycloak = KeycloakOrgEntityProvider.fromConfig(config, {
+      id: 'development',
+      logger: getVoidLogger(),
+      schedule: 'manual',
+    });
+
+    const result = keycloak.map(async k => await k.connect(connection));
+
+    expect(result).toHaveLength(1);
```
Comment (invincibleJai) [inline plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.test.ts:82]: `keycloak.map(async k => await k.connect(connection));` will return list of promises `[ Promise { <pending> } ]` and length of result will always be same as keycloak

and if await for promises to be resolved it gives `[undefined]`

File: plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.test.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  });
+
+  it('should not read without a connection', async () => {
```
Comment (invincibleJai) [inline plugins/keycloak-backend/src/providers/KeycloakOrgEntityProvider.test.ts:85]:
```suggestion
  it('should not read without a connection', () => {
```

File: plugins/keycloak-backend/__fixtures__/helpers.ts (hunk 2 above)
Comment (schultzp2020) [reply plugins/keycloak-backend/__fixtures__/helpers.ts:1]: https://github.com/janus-idp/backstage-plugins/issues/445

File: plugins/keycloak-backend/src/lib/read.test.ts (hunk 1 above)
Comment (schultzp2020) [reply plugins/keycloak-backend/src/lib/read.test.ts:9]: Sure, I'll update the object with the required types since there are only 3 of them.

**Review by invincibleJai — APPROVED** [review]
LGTM

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**